### PR TITLE
Add some tests for role rule fns

### PR DESCRIPTION
**Pull request by cap10morgan**
This test fails on db main HEAD, but is fixed by https://github.com/fluree/db/pull/199.

### CODE REVIEW

**Review by zonotope — APPROVED** [review]
lgtm.

File: test/fluree/db/ledger/rule_functions_test.clj [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+          zsmith-keys        (test/load-keys "zsmith-auth")
+          ledger             (test/rand-ledger "test/role-fn-update")
+          _                  (assert-success (test/transact-schema ledger "chat.edn" :clj))
+          _                  (assert-success (test/transact-schema ledger "chat-preds.edn" :clj))
+          _                  (assert-success (test/transact-data ledger "chat.edn" :clj))
+          {:keys [block]} (assert-success (test/transact-data ledger "chat-rules.edn" :clj))
+
+          db                 (fdb/db (:conn test/system) ledger {:syncTo block})
+
+          ;; Can edit own chat messages
```
[the remark this answers is not in the document]
Comment (cap10morgan) [reply test/fluree/db/ledger/rule_functions_test.clj:32]: Can you nest those?

[the remark this answers is not in the document]
Comment (zonotope) [reply test/fluree/db/ledger/rule_functions_test.clj:32]: yep: https://github.com/fluree/db/blob/main/test/fluree/db/query/sql_test.cljc

**Comment by cap10morgan** [comment]
Thanks for the review, @zonotope!

**Pull request merged**